src/target/riscv/riscv.c

```diff
@@ -655,20 +664,41 @@ static int set_trigger(struct target *target, unsigned int idx, riscv_reg_t tdat
 		return ERROR_FAIL;
 	if (riscv_reg_get(target, &tdata2_rb, GDB_REGNO_TDATA2) != ERROR_OK)
 		return ERROR_FAIL;
-	bool tdata1_config_denied = (tdata1 & ~tdata1_ignore_mask) != (tdata1_rb & ~tdata1_ignore_mask);
+	bool unsupported_napot_range = false;
+	bool tdata1_config_denied = true;
+	const uint32_t type = get_field(tdata1, CSR_TDATA1_TYPE(riscv_xlen(target)));
+	const bool is_mcontrol = type == CSR_TDATA1_TYPE_MCONTROL;
+	const bool is_napot_match = get_field(tdata1, CSR_MCONTROL_MATCH) == CSR_MCONTROL_MATCH_NAPOT;
+	if (is_mcontrol && is_napot_match) {
+		maskmax = get_field(tdata1_rb, CSR_MCONTROL_MASKMAX(riscv_xlen(target)));
+		const unsigned int napot_size = count_trailing_ones(tdata2) + 1;
+		if (maskmax < napot_size)
+			unsupported_napot_range = true;
+		tdata1_config_denied = (tdata1 & ~maskmax) != (tdata1_rb & ~maskmax);
```

Comment:
While refactoring the code, a bug was introduced: There would need to be separate variables for `maskmax_value` and `maskmax_mask`, as shown below.

Instead of just fixing it, please, improve the overall structure of the code as shown in the below comment.

```c
		maskmax_mask = CSR_MCONTROL_MASKMAX(riscv_xlen(target));
		maskmax_value = get_field(tdata1_rb, maskmax_mask);
		const unsigned int napot_size = count_trailing_ones(tdata2) + 1;
		if (maskmax_value < napot_size)
			unsupported_napot_range = true;
		tdata1_config_denied = (tdata1 & ~maskmax_mask) != (tdata1_rb & ~maskmax_mask);
```

Reply:
thanks a lot, before:line：677 value `maskmax` is a bug